modules/programs/mergiraf.nix

```diff
@@ -0,0 +1,35 @@
+{ pkgs, config, lib, ... }:
+let
+  inherit (lib)
+    mkEnableOption mkPackageOption types literalExpression mkIf maintainers;
+  cfg = config.programs.mergiraf;
+  mergiraf = "${cfg.package}/bin/mergiraf";
+in {
+  meta.maintainers = [ maintainers.bobvanderlinden ];
+
+  options = {
+    programs.mergiraf = {
+      enable = mkEnableOption "mergiraf";
+      package = mkPackageOption pkgs "mergiraf" { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    programs.git = {
+      extraConfig = {
+        merge.mergiraf = {
+          name = "mergiraf";
+          driver =
+            "${mergiraf} merge --git %O %A %B -s %S -x %X -y %Y -p %P -l %L";
+        };
+      };
+    };
+
+    xdg.configFile."git/attributes".source =
+      pkgs.runCommand "mergiraf-git-attributes" { } ''
+        ${mergiraf} languages --gitattributes > $out
+      '';
```

Comment:
This is something I'm unsure of. Right now, it'll conflict if someone has set `programs.git.attributes`. It does update all language settings automatically.

The alternative would be hard-coding the languages, but we need to keep this list up-to-date, which is less than ideal.

Another alternative is allowing `programs.git.attributes` to also include other files and concat it all. I'm unsure whether this is wanted for now.

Reply:
Definitely shouldn't be overriding the source contents of `git/attributes` from another module. We should expose an option in the git module to support appending to the generated config.

Reply:
I agree, that makes sense. I've now opted to just handle all file extensions. This avoids the need to write the output of mergiraf to `~/.config/git/attributes`, so this is not an issue anymore :+1: